test/dag_test.go

```diff
@@ -81,110 +74,56 @@ spec:
     # Sleep for N seconds so that we can check that tasks that
     # should be run in parallel have overlap.
     script: 'sleep %d'
-  - image: busybox
-    script: 'ln -s $(resources.inputs.repo.path) $(resources.outputs.repo.path)'
 `, helpers.ObjectNameForTest(t), namespace, int(sleepDuration.Seconds())))
 	if _, err := c.V1beta1TaskClient.Create(ctx, echoTask, metav1.CreateOptions{}); err != nil {
 		t.Fatalf("Failed to create echo Task: %s", err)
 	}
 
-	// Create the repo PipelineResource (doesn't really matter which repo we use)
-	repoResource := parse.MustParsePipelineResource(t, fmt.Sprintf(`
-metadata:
-  name: %s
-spec:
-  type: git
-  params:
-  - name: Url
-    value: https://github.com/githubtraining/example-basic
-`, helpers.ObjectNameForTest(t)))
-	if _, err := c.V1alpha1PipelineResourceClient.Create(ctx, repoResource, metav1.CreateOptions{}); err != nil {
-		t.Fatalf("Failed to create simple repo PipelineResource: %s", err)
-	}
-
 	// Intentionally declaring Tasks in a mixed up order to ensure the order
 	// of execution isn't at all dependent on the order they are declared in
 	pipeline := parse.MustParseV1beta1Pipeline(t, fmt.Sprintf(`
 metadata:
   name: %s
   namespace: %s
 spec:
-  resources:
-  - name: repo
-    type: git
   tasks:
   - name: pipeline-task-3
     params:
     - name: text
       value: wow
-    resources:
-      inputs:
-      - from:
-        - pipeline-task-2-parallel-1
-        - pipeline-task-2-parallel-2
-        name: repo
-        resource: repo
-      outputs:
-      - name: repo
-        resource: repo
     taskRef:
       name: %s
+    runAfter:
```

Comment:
NIT: I think an inplicit type of DAG ordering (e.g. param -> result) would match better with the original test as opposed to an explicit one like runAfter. 
That said, without an e2e coverage report is hard to say.
Once we switch to golang 1.20, we could benefit from https://go.dev/testing/coverage/

Reply:
Thanks, I replaced the tasks to be using result and params so as to be implicit.